apple/Brushes/RNSVGPainter.mm

```diff
@@ -232,6 +232,15 @@ - (void)paintRadialGradient:(CGContextRef)context bounds:(CGRect)bounds
   CGFloat rx = [self getVal:[_points objectAtIndex:2] relative:width];
   CGFloat ry = [self getVal:[_points objectAtIndex:3] relative:height];
 
+  if (rx <= 0 || ry <= 0) {
+    // Gradient with radius = 0 should be rendered as solid color of the last stop
+    rx = width;
+    ry = height;
+    CGGradientRelease(gradient);
+    NSArray<NSNumber *> *gradientArray = @[_colors.firstObject, _colors.lastObject, _colors[_colors.count-2], _colors.lastObject];
```

Comment:
Could you quickly explain what each element of this array corresponds to? And since we have `_colors.count-2`, is there a case there aren't 3 items in the `_colors` array?

Reply:
Sure! When `radialGradient`'s `r=0` it should be painted as a single color using the color of the last `stop`.

The `_colors` array corresponds to the `Stop` elements inside `RadialGradient`. It consists of `offset` and `stopColor` respectively so the `gradientArray` should look like this

`[offset from first, color from last, offset from last, color from last]`

In addition, above is a check that `_colors.count` is not 0 and given that it should always consists of an even number of elements (offset, color), even for `stop` with no props the default values should be passed.

Reply:
Perfect, thanks for the explanation!